src/test/jtx/mpt.h

```diff
@@ -0,0 +1,267 @@
+{
+    std::optional<std::uint64_t> maxAmt = std::nullopt;
+    std::optional<std::uint8_t> assetScale = std::nullopt;
+    std::optional<std::uint16_t> transferFee = std::nullopt;
+    std::optional<std::string> metadata = std::nullopt;
+    std::optional<std::uint32_t> ownerCount = std::nullopt;
+    std::optional<std::uint32_t> holderCount = std::nullopt;
+    bool fund = true;
+    std::optional<std::uint32_t> flags = {0};
+    std::optional<TER> err = std::nullopt;
+};
+
+struct MPTDestroy
+{
+    AccountP issuer = nullptr;
+    std::optional<MPTID> id = std::nullopt;
+    std::optional<std::uint32_t> ownerCount = std::nullopt;
+    std::optional<std::uint32_t> holderCount = std::nullopt;
+    std::optional<std::uint32_t> flags = std::nullopt;
+    std::optional<TER> err = std::nullopt;
+};
+
+struct MPTAuthorize
+{
+    AccountP account = nullptr;
+    AccountP holder = nullptr;
+    std::optional<MPTID> id = std::nullopt;
+    std::optional<std::uint32_t> ownerCount = std::nullopt;
+    std::optional<std::uint32_t> holderCount = std::nullopt;
+    std::optional<std::uint32_t> flags = std::nullopt;
+    std::optional<TER> err = std::nullopt;
+};
+
+struct MPTSet
+{
+    AccountP account = nullptr;
+    AccountP holder = nullptr;
+    std::optional<MPTID> id = std::nullopt;
+    std::optional<std::uint32_t> ownerCount = std::nullopt;
+    std::optional<std::uint32_t> holderCount = std::nullopt;
+    std::optional<std::uint32_t> flags = std::nullopt;
+    std::optional<TER> err = std::nullopt;
+};
+
+class MPTTester
+{
+    Env& env_;
+    Account const& issuer_;
+    std::unordered_map<std::string, AccountP> const holders_;
+    std::optional<MPTID> id_;
+    std::optional<uint256> issuanceKey_;
+    bool close_;
+
+public:
+    MPTTester(Env& env, Account const& issuer, MPTConstr const& constr = {});
+
+    void
+    create(MPTCreate const& arg = MPTCreate{});
+
+    void
+    destroy(MPTDestroy const& arg = MPTDestroy{});
+
+    void
+    authorize(MPTAuthorize const& arg = MPTAuthorize{});
+
+    void
+    set(MPTSet const& set = {});
+
+    [[nodiscard]] bool
+    checkMPTokenAmount(Account const& holder, std::int64_t expectedAmount)
+        const;
+
+    [[nodiscard]] bool
+    checkMPTokenOutstandingAmount(std::int64_t expectedAmount) const;
+
+    [[nodiscard]] bool
+    checkFlags(uint32_t const expectedFlags, AccountP holder = nullptr) const;
+
+    Account const&
+    issuer() const
+    {
+        return issuer_;
+    }
+    Account const&
+    holder(std::string const& h) const;
+
+    void
+    pay(Account const& src,
+        Account const& dest,
+        std::int64_t amount,
+        std::optional<TER> err = std::nullopt);
+
+    void
+    claw(
+        Account const& issuer,
+        Account const& holder,
+        std::int64_t amount,
+        std::optional<TER> err = std::nullopt);
+
+    PrettyAmount
+    mpt(std::int64_t amount) const;
+
+    uint256 const&
+    issuanceKey() const
+    {
+        assert(issuanceKey_);
```

Comment:
I don't generally like `assert`s in unit tests.
```suggestion
        if (!env_.test.BEAST_EXPECT(issuanceKey_))
            Throw<std::logic_error>("Uninitialized issuanceKey");
```